components/cli/pkg/commands/build.go

```diff
@@ -19,323 +19,131 @@
 package commands
 
 import (
-	"bufio"
-	"bytes"
 	"encoding/json"
-	"errors"
 	"fmt"
 	"io/ioutil"
 	"os"
-	"os/exec"
-	"os/user"
 	"path"
 	"path/filepath"
-	"regexp"
 	"strings"
 	"time"
 
-	"github.com/cellery-io/sdk/components/cli/pkg/image"
-
 	"github.com/ghodss/yaml"
 
+	"github.com/cellery-io/sdk/components/cli/ballerina"
+	"github.com/cellery-io/sdk/components/cli/cli"
 	"github.com/cellery-io/sdk/components/cli/pkg/constants"
+	"github.com/cellery-io/sdk/components/cli/pkg/image"
 	"github.com/cellery-io/sdk/components/cli/pkg/util"
 	"github.com/cellery-io/sdk/components/cli/pkg/version"
 )
 
 // RunBuild executes the cell's build life cycle method and saves the generated cell image to the local repo.
 // This also copies the relevant ballerina files to the ballerina repo directory.
-func RunBuild(tag string, fileName string) {
-	fileExist, err := util.FileExists(fileName)
-	if !fileExist {
-		util.ExitWithErrorMessage("Unable to build image",
-			errors.New(fmt.Sprintf("file '%s' does not exist", util.Bold(fileName))))
-	}
-
-	parsedCellImage, err := image.ParseImageTag(tag)
-	if err != nil {
-		util.ExitWithErrorMessage("Error occurred while parsing image", err)
-	}
-	repoLocation := filepath.Join(util.UserHomeDir(), constants.CELLERY_HOME, "repo", parsedCellImage.Organization,
-		parsedCellImage.ImageName, parsedCellImage.ImageVersion)
-
-	spinner := util.StartNewSpinner("Building image " + util.Bold(tag))
-	defer func() {
-		spinner.Stop(true)
-	}()
-
-	// First clean target directory if exists
-	projectDir, err := os.Getwd()
-	targetDir := filepath.Join(projectDir, "target")
-	if err != nil {
-		spinner.Stop(false)
-		util.ExitWithErrorMessage("Error in getting current directory location", err)
-	}
-	_ = os.RemoveAll(targetDir)
-
+func RunBuild(cli cli.Cli, tag string, fileName string) error {
+	var err error
+	var projectDir string
+	var parsedCellImage *image.CellImage
+	var iName []byte
+	if projectDir, err = cli.FileSystem().CurrentDir(); err != nil {
+		return err
+	}
+	if parsedCellImage, err = image.ParseImageTag(tag); err != nil {
+		return fmt.Errorf("error occurred while parsing image, %v", err)
+	}
+	fmt.Println(parsedCellImage)
```

Comment:
Please remove print and use cli interface

Reply:
Removed.